psh/pshapp/env.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Phoenix-RTOS
+ *
+ * psh variables and environment
+ *
+ * implements psh builtins:
+ * - export: set and propagate variable names to environment
+ * - unset:  removes variables whose names match
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+#include "../psh.h"
+
+
+extern char **environ;
+
+
+static int psh_validateVarName(const char *name, int len)
+{
+	if ((name == NULL) || (*name == '\0') || (isdigit(*name) != 0)) {
+		return -EINVAL;
+	}
+
+	/* len<0 means: nevermind 'len' and validate until '\0' */
```

Comment:
Validate until token passed (i.e. `'='`) might be more useful than `len` - we wouldn't need to use `strchr()`

Reply:
There are several cases where without strchr, the psh code would increase unnecessarily, now it checks correctly following cases:
`export 1INVALID = VALID123 _VALID=1234 =INVALID =ALSOINVALID=VALUE VALID=VALUE`
Invalid are: `1INVALID` `=` `=INVALID` `=ALSOINVALID=VALUE`
It is prepared to work with quotes in other pull-request.